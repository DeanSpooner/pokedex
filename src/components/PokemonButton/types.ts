export interface Props {
  dexNumber: number;
  pokeName: string;
  spritePath?: string;
}
