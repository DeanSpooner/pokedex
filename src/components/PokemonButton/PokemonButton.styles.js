import styled from "styled-components";

export const MonsterButton = styled.button`
  align-items: center;
  grid-column: auto;
  background-color: lightgray;
  border-radius: 12px;
  width: 150px;
  height: 150px;
  font-size: 16px;
  margin: 6px;
  :hover {
    background-color: lightgreen;
    cursor: pointer;
  }
`;

export const PokemonImg = styled.img`
  margin: auto;
  display: block;
`;
