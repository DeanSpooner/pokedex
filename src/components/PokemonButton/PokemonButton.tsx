import * as React from "react";
import { Props } from "./types";
import { useRouter } from "next/router";

import { MonsterButton, PokemonImg } from "./PokemonButton.styles";

const PokemonButton: React.FC<Props> = ({
  dexNumber,
  pokeName,
  spritePath,
}) => {
  const router = useRouter();

  const clickHandler = () => {
    router.push(
      dexNumber < 152 ? `/pokemon/${dexNumber}` : `/newerpokemon/${dexNumber}`
    );
  };

  return (
    <MonsterButton onClick={clickHandler} data-testid="MonsterButton">
      <>
        {spritePath && <PokemonImg src={spritePath} />}#{dexNumber}: {pokeName}
      </>
    </MonsterButton>
  );
};

export default PokemonButton;
