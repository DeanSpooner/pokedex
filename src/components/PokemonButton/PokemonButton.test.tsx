/** @jest-environment jsdom */

import React from "react";
import { render } from "@testing-library/react";
import PokemonButton from "./PokemonButton";

describe("MonsterButton", () => {
  it("matches snapshot", () => {
    const { getByTestId } = render(
      <PokemonButton
        pokeName={"Pikachu"}
        dexNumber={25}
        spritePath={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${25}.png`}
      />
    );
    expect(getByTestId("MonsterButton")).toMatchSnapshot();
  });
});
