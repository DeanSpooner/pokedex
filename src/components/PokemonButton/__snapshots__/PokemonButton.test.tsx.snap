// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterButton matches snapshot 1`] = `
<button
  class="PokemonButtonstyles__MonsterButton-sc-18altu4-0 gRLgdG"
  data-testid="MonsterButton"
>
  <img
    class="PokemonButtonstyles__PokemonImg-sc-18altu4-1 cmJZMa"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
  />
  #
  25
  : 
  Pikachu
</button>
`;
